autogpt_platform/backend/backend/blocks/example/_api.py

```diff
@@ -0,0 +1,128 @@
+"""
+API module for Example API integration.
+
+This module provides a example of how to create a client for an API.
+"""
+
+# We also have a Json Wrapper library available in backend.util.json
+from json import JSONDecodeError
+from typing import Any, Dict, Optional
+
+from backend.data.model import APIKeyCredentials
+
+# This is a wrapper around the requests library that is used to make API requests.
+from backend.util.request import Requests
+
+
+class ExampleAPIException(Exception):
+    def __init__(self, message: str, status_code: int):
+        super().__init__(message)
+        self.status_code = status_code
+
+
+class ExampleClient:
```

Comment:
Why is this client boilerplate needed?

Reply:
The purpose of this PR is capture the best practices of adding a new integration block. Creating a client seems to be how we have implemented this with other bocks. 

Do you feel this is not best practice? 
If so what would you suggest?

Reply:
Yes I think it makes sense when the SDK support is missing.

But stuff here getting copied over and over without any refactoring: these ended up being duplicated in the HTTP client part:
* Setting up Request class
* Set content type, passing header to Authorization
* Adding to trusted origin (which is not actually needed)

Other issues:
* Return the value of Dict which should be properly typed. (also the seems like our convention is using dict/list | None as opposed to Dict List Optional)

* Creating a custom exception, then catching it within the same class, shouldn't this be the other way, catching an HTTP error and wrapping it to the custom exception (if it's even needed)?

Reply:
we can use pydantic objects and that should be preferred except when there's a fuck off big response from the API. That case should be model dumpted to a "dict" and attached as a raw object. Github Apollo/smartlead do this